src/client.rs

```diff
@@ -1452,6 +1452,17 @@ impl<C: Config> Client<C> {
                 let entity = entities
                     .get(id)
                     .expect("entity IDs in spatial index should be valid at this point");
+
+                // Skip spawning players not in the player list because they would be invisible
+                // otherwise.
+                if entity.kind() == EntityKind::Player {
+                    if let Some(list_id) = &self.player_list {
+                        player_lists.get(list_id).entry(entity.uuid())?;
```

Comment:
Correct me if I'm wrong, but doesn't this line mean, that we would return the player entity if it is found? Then we would skip the spawn logic below (1470-1576)?

Reply:
These lines say:
1. If the entity is a player, then...
2. If the client has a player list, then...
3. Get the entry with the entity's UUID. If the value is `Some` (the entry exists), do nothing. Otherwise, return early with `None` (the entry does not exist).